docs/okeps/okep-5088-evpn.md

```diff
@@ -0,0 +1,789 @@
+the netdevs and finding Linux Bridges/VLANs within the node.
+
+For IP-VRFs, we need to be able to add the following configuration:
+
+```bash
+vrf udnA
+ vni 100
+exit-vrf
+!
+router bgp 64512 vrf udnA
+ !
+ address-family l2vpn evpn
+  route-target import 64512:100
+  route-target export 64512:100
+ exit-address-family
+exit 
+```
+
+In the above configuration, the "vrf udnA" refers to an IP-VRF, then we need a router bgp vrf stanza activating EVPN
+for it. EVPN also utilizes BGP Route-Targets (RT) in order to import/export routes between IP-VRFs. We must configure at
+least the targets for the VRF itself so that it will import/export routes from other nodes in its own IP-VRF. In the
+future this route-target config may be extended to include other IP-VRFs (UDNs) in order to connect networks together.
+Similar to the "Network Connect" feature to connect UDNs via OVN+Geneve.
+
+How the FRR-K8S API will be extended to accommodate these configurations will be documented later. <TODO Jaime>
+
+### VTEP CRD
+
+A VTEP CRD will be created which allows an admin to define VTEP IPs to be associated with an EVPN enabled CUDN. In the
+the future this VTEP CRD may be extended to other use cases like providing VTEP IPs for Geneve tunnels, etc.
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: VTEP
+metadata:
+  name: evpn-vtep
+spec:
+  cidr: 100.64.0.0/24
+  mode: managed
+```
+
+The cidr field is mandatory. If the mode is "managed", then OVN-Kubernetes will handle allocating and assigning
+VTEP IPs per node. If the mode is not provided, or is "unmanaged", then it is left to some other provider to handle
+adding the IP address to each node from the subnet provided. In unmanaged mode, OVN-Kubernetes will find an interface on
+the node which has an IP address in the cidr, and use that IP address. Unmanaged mode may be preferred where a provider
+handles assigning VTEP IPs within its EVPN fabric. In this case OVN-Kubernetes cluster is integrating into an already
+configured EVPN fabric. Therefore VTEP IP provisioning may be done by the provider and configured for each node.
+
+The IP address assigned or found by OVN-Kubernetes will be annotated to the node as "k8s.ovn.org/<vtep name>: <ip>".
+If all nodes do not have an IP address annotated for this VTEP, the VTEP CR will go into error state.
+
+### CUDN CRD changes for EVPN
+
+This API change depends on the [No Overlay Feature](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5289) which will
+introduce a new Transport field that we can extend to use EVPN. The API change will look like this:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: l3-primary
+  labels:
+    bgp: enabled
+spec:
+  namespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: udn-test
+  network:
+    topology: Layer2
+    layer2:
+      role: Primary
+      subnets:
+      - 10.20.100.0/16
```

Comment:
@trozet we are going to have the transit router in place for layer2 before evpn is implemented right ? because if not the defautl gw reconciliation for VMs is kind of involve since we don't have gateway routers at OVN anymore.